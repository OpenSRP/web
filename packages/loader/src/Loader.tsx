import React from 'react';
import { Ripple } from '@onaio/loaders';

/** interface for Ripple props */
export interface LoaderProps {
  borderColor?: string;
  borderStyle?: string;
  borderWidth?: string;
  height?: string;
  minHeight?: string;
  width?: string;
}

export const defaultProps: Partial<LoaderProps> = {
  borderColor: '#ff5d00',
  borderStyle: 'solid',
  borderWidth: '4px',
  height: '64px',
  minHeight: '80vh',
  width: '64px',
};

/** Loading component that displays a nice ripple
 *
 * @param {object} props should contain the UI styles.
 * @returns {Element} - actions
 */
const Loader: React.FC<LoaderProps> = (props: LoaderProps) => <Ripple {...props} />;

Loader.defaultProps = defaultProps;

export { Loader };
