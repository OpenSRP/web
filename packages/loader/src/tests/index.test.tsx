import { mount } from 'enzyme';
import React from 'react';

import { Loader } from '../Loader';

describe('components/Loaders/Ripple', () => {
  it('renders the Loader component', () => {
    const wrapper = mount(<Loader />);
    expect(wrapper.props()).toMatchSnapshot();
  });

  it('renders the Loader component when props are given', () => {
    const props = {
      borderColor: '#ff0000',
      borderStyle: 'solid',
      borderWidth: '6px',
      height: '128px',
      minHeight: '70vh',
      width: '128px',
    };
    const wrapper = mount(<Loader {...props} />);
    expect(wrapper.props()).toMatchSnapshot();
  });
});
