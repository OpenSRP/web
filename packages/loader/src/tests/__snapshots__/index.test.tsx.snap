// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Loaders/Ripple renders the Loader component 1`] = `
Object {
  "borderColor": "#ff5d00",
  "borderStyle": "solid",
  "borderWidth": "4px",
  "height": "64px",
  "minHeight": "80vh",
  "width": "64px",
}
`;

exports[`components/Loaders/Ripple renders the Loader component when props are given 1`] = `
Object {
  "borderColor": "#ff0000",
  "borderStyle": "solid",
  "borderWidth": "6px",
  "height": "128px",
  "minHeight": "70vh",
  "width": "128px",
}
`;
